bool is_start
---
bool is_received
